libraries/lib-wave-track/tests/AudioSegmentSampleViewTest.cpp

```diff
@@ -0,0 +1,57 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegmentSampleViewTest.cpp
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#include "AudioSegmentSampleView.h"
+
+#include <catch2/catch.hpp>
+
+TEST_CASE("AudioSegmentSampleView", "Copy returns expected values when")
+{
+   SECTION("AudioSegmentSampleView is silent")
+   {
+      AudioSegmentSampleView sut { 3 };
+      std::vector<float> buffer(4);
+      std::fill(buffer.begin(), buffer.end(), 1.f);
+      SECTION("and asked MORE values than it holds.")
+      {
+         REQUIRE(sut.Copy(buffer.data(), 4u) == 3u);
+         REQUIRE(buffer == std::vector<float> { 0.f, 0.f, 0.f, 1.f });
+      }
+      SECTION("and asked LESS values than it holds.")
```

Comment:
I like the improved test!

My last irritating trivial comment for today:  "FEWER" not "LESS" is better English according to the grammar books, but a lot of English speakers don't observe that usage.